spec/spec_helper.rb

```diff
@@ -194,6 +194,40 @@ def use_headful_chrome? = ENV.fetch('HEADFUL_CHROME', nil).present?
     RSpec::Support::ObjectFormatter.default_instance.max_formatted_output_length = 2_000
   end
 
+  config.around(:each, type: :feature) do |example|
+    $test_log_string_io ||= StringIO.new
+    $test_log_string_io.reopen
+
+    string_io_logger = ActiveSupport::Logger.new($test_log_string_io)
+    string_io_logger.formatter = Rails.logger.formatter.dup
+    string_io_logger.level = :debug
+
+    Rails.with(logger: string_io_logger) do
+      Sidekiq.default_configuration.with(logger: string_io_logger) do
+        # We can't just do ActiveRecord::Base.with(...) because `with` is an ActiveRecord method.
+        Object.instance_method(:with).bind_call(ActiveRecord::Base, logger: string_io_logger) do
```

Comment:
For some Rails stuff (like ActionCable and ActionController), they seem to use the `Rails.logger`, and so changing that causes them to log to our StringIO. However, Sidekiq and ActiveRecord seem to create their own loggers (maybe initially based partially on `Rails.logger`, I'm not sure), and so changing `Rails.logger` doesn't affect their behavior. That's why here we are also adjusting the Sidekiq and ActiveRecord loggers, specifically.